dcsazure_AzureSQL_to_AzureSQL_mask_pl/README.md

```diff
@@ -0,0 +1,111 @@
+
+`Azure SQL` (metadata) - Linked service associated with your hosted metadata store. This will be used for the following
+steps:
+* dcsazure_AzureSQL_to_AzureSQL_metadata_mask_ds (Azure SQL Database dataset)
+* dcsazure_AzureSQL_to_AzureSQL_mask_params_df/Ruleset (dataFlow)
+* dcsazure_AzureSQL_to_AzureSQL_mask_params_df/TypeMapping (dataFlow)
+
+`REST` (DCS for Azure) - Linked service associated with calling DCS for Azure. This will be used for the following
+steps:
+* dcsazure_AzureSQL_to_AzureSQL_mask_df (dataFlow)
+
+### How It Works
+* Select Tables Without Required Masking. This is done by querying the metadata store.
+  * Filter If Copy Unmask Enabled. This is done by applying a filter based on the value of `P_COPY_UNMASKED_TABLES`
+    * For Each Table With No Masking. Provided we have any rows left after applying the filter
+      * Get Sink Table Details No Masking. Query AzureSQL for sink table details
+      * Get Sink Table Metadata No Masking. Query Metadata store to construct metadata information for sink table
+      * If Copy Write Supported. Evaluate whether we can copy data to the sink table
+        * If true, the writer version is supported, copy data by calling
+          `dcsazure_AzureSQL_to_AzureSQL_copy_df`
+        * If false, the writer version is unsupported, fail the pipeline
+* Select Tables That Require Masking. This is done by querying the metadata store. This will provided a list of tables
+  that need masking, and if they need to be masked leveraging conditional algorithms, the set of required filters.
+  * For Each Table To Mask
+    * For each table that requires masking:
+      * Get the source table's metadata
+      * Get the sink table's details
+        * If the table can be written to (i.e. the Data Factory backend supports this table writer version)
+          * Get the metadata needed for the sink table
+          * If no filter needs to be applied:
+            * Call the `dcsazure_AzureSQL_to_AzureSQL_unfiltered_mask_params_df` data flow to generate masking
+              parameters
+            * Call the `dcsazure_AzureSQL_to_AzureSQL_unfiltered_mask_df` data flow, passing in parameters as
+              generated by the generate masking parameters dataflow
+          * If a filter needs to be applied:
+            * Call the `dcsazure_AzureSQL_to_AzureSQL_filtered_mask_params_df` data flow to generate masking
+              parameters using the filter alias
+            * Call the `dcsazure_AzureSQL_to_AzureSQL_filtered_mask_df` data flow, passing in parameters as
+              generated by the generate masking parameters dataflow and the filter as determined by the output of For
+              Each Table To Mask
+        * If the table cannot be written to (i.e. the Data Factory backend doesn't support this table writer version)
+          * Provide a `400` status error code to indicate the failure of this table (with the appropriate message)
```

Comment:
```suggestion
* Select Tables That Require Masking. This is done by querying the metadata store. This will provide a list of tables
  that need masking, and if they need to be masked leveraging conditional algorithms, the set of required filters.
  * For Each Table To Mask
    * Check if the table must be masked with a filter condition
      * If no filter needs to be applied:
        * Call the `dcsazure_AzureSQL_to_AzureSQL_unfiltered_mask_params_df` data flow to generate masking parameters
        * Call the `dcsazure_AzureSQL_to_AzureSQL_unfiltered_mask_df` data flow, passing in parameters as generated by
          the generate masking parameters dataflow
        * Update the mapped status based on the success of this dataflow, and fail accordingly
      * If a filter needs to be applied:
        * Call the `dcsazure_AzureSQL_to_AzureSQL_filtered_mask_params_df` data flow to generate masking parameters
          using the filter alias
        * Call the `dcsazure_AzureSQL_to_AzureSQL_filtered_mask_df` data flow, passing in parameters as generated by
          the generate masking parameters dataflow and the filter as determined by the output of For Each Table To Mask
        * Update the mapped status based on the success of this dataflow, and fail accordingly
```